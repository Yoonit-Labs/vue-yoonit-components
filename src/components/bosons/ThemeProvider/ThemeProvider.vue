 <template lang="pug">
  .theme-provider(
    :data-composition="composition"
    :data-theme="theme"
  )
    slot
</template>

<script>
/**
 * ██╗   ██╗ ██████╗  ██████╗ ███╗   ██╗██╗████████╗
 * ╚██╗ ██╔╝██╔═══██╗██╔═══██╗████╗  ██║██║╚══██╔══╝
 *  ╚████╔╝ ██║   ██║██║   ██║██╔██╗ ██║██║   ██║
 *   ╚██╔╝  ██║   ██║██║   ██║██║╚██╗██║██║   ██║
 *    ██║   ╚██████╔╝╚██████╔╝██║ ╚████║██║   ██║
 *    ╚═╝    ╚═════╝  ╚═════╝ ╚═╝  ╚═══╝╚═╝   ╚═╝
 *
 * https://yoonit.dev - about@yoonit.dev
 *
 * Vue Yoonit Components
 * VueJS Atomic Design System framework
 *
 * Vitória Costa, Tiago Brito, Fernando Junior, Sabrina Sampaio, Gabriel Mule, Gabriel Moraes, Gabriel Rizzo & Luigui Delyer @ 2020-2021
 */

export default {
  name: 'ThemeProvider',

  props: {
    composition: {
      type: String,
      default: 'default'
    },
    theme: {
      type: String,
      default: 'default'
    }
  }
}
</script>
