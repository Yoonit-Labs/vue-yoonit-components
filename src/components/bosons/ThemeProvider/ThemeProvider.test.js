import { shallowMount } from '@vue/test-utils'
import ThemeProvider from '@/components/bosons/ThemeProvider/ThemeProvider.vue'

const classBlock = 'theme-provider'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(ThemeProvider, {
    slots: { default: SlotText }
  })
}

describe('ThemeProvider Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.find('.theme-provider').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('composition', () => {
      it('Has a valid default value', () => {
        expect(ThemeProvider.props.composition.default).toBe('default')
      })
      it('Includes data composition', async () => {
        await wrapper.setProps({ composition: 'blue' })
        expect(wrapper.find(`.${classBlock}`).attributes('data-composition')).toEqual('blue')
      })
    })
    describe('theme', () => {
      it('Has a valid default value', () => {
        expect(ThemeProvider.props.theme.default).toBe('default')
      })
      it('Includes data composition', async () => {
        await wrapper.setProps({ theme: 'dark' })
        expect(wrapper.find(`.${classBlock}`).attributes('data-theme')).toEqual('dark')
      })
    })
  }) // describe Props
})
