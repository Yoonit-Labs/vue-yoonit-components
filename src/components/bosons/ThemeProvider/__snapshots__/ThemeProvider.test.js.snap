// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider Component Matches Snapshot 1`] = `<div data-composition="default" data-theme="default" class="theme-provider">Default Slot Text</div>`;
